src/csharp.tmLanguage.yml

```diff
@@ -2343,11 +2390,14 @@ repository:
         patterns:
         - include: '#type'
 
-  this-or-base-expression:
-    match: \b(?:(base)|(this))\b
-    captures:
-      '1': { name: keyword.other.base.cs }
-      '2': { name: keyword.other.this.cs }
+  language-variable:
+    patterns:
+    # - name: variable.language.$1.cs
+    #   match: \b(base|this)\b
+    - name: keyword.other.$1.cs
+      match: \b(base|this)\b
```

Comment:
Currently `this` and `base` are scoped to `keyword.other` (which I think is overused).
The convention/recommendation for this/self keywords seem to be `variable.language` however.
What do you think?

Reply:
[macromates](https://macromates.com/manual/en/language_grammars) agrees with your assessment. I think we should make the switch.

<img width="703" alt="image" src="https://github.com/dotnet/csharp-tmLanguage/assets/611219/33dfcd36-bada-4730-92fa-3e235df4906c">